portallocator/portallocator.go

```diff
@@ -165,6 +165,14 @@ func (p *PortAllocator) ReleasePort(ip net.IP, proto string, port int) error {
 	if !ok {
 		return nil
 	}
+	for _, pr := range protomap[proto].portRanges {
```

Comment:
I don't think so. The port ranges that are created by users maybe overlapped.

No. See
```
 if port >= pr.begin && port <= pr.last
```